pinot-common/src/main/java/org/apache/pinot/common/config/TagNameUtils.java

```diff
@@ -27,78 +28,93 @@ private TagNameUtils() {
   }
 
   public final static String DEFAULT_TENANT_NAME = "DefaultTenant";
+  private final static String BROKER_TAG_SUFFIX = "_" + TenantRole.BROKER;
+  private final static String OFFLINE_SERVER_TAG_SUFFIX = "_" + ServerType.OFFLINE;
+  private final static String REALTIME_SERVER_TAG_SUFFIX = "_" + ServerType.REALTIME;
 
-  private static String buildRealtimeTagFromTenantName(String tenantName) {
-    return tenantName + "_" + ServerType.REALTIME.toString();
-  }
-
-  private static String buildOfflineTagFromTenantName(String tenantName) {
-    return tenantName + "_" + ServerType.OFFLINE.toString();
-  }
-
-  private static String buildBrokerTenantTagFromTenantName(String tenantName) {
-    return tenantName + "_" + TenantRole.BROKER.toString();
+  public static boolean isBrokerTag(String tagName) {
+    return tagName.endsWith(BROKER_TAG_SUFFIX);
   }
 
   public static boolean isServerTag(String tagName) {
     return isOfflineServerTag(tagName) || isRealtimeServerTag(tagName);
   }
 
   public static boolean isOfflineServerTag(String tagName) {
-    return tagName.endsWith(ServerType.OFFLINE.toString());
+    return tagName.endsWith(OFFLINE_SERVER_TAG_SUFFIX);
   }
 
   public static boolean isRealtimeServerTag(String tagName) {
-    return tagName.endsWith(ServerType.REALTIME.toString());
+    return tagName.endsWith(REALTIME_SERVER_TAG_SUFFIX);
   }
 
-  public static boolean isBrokerTag(String tagName) {
-    return tagName.endsWith(TenantRole.BROKER.toString());
+  public static String getTenantFromTag(String tagName) {
```

Comment:
You can call `isOfflineServerTag`, `isRealtimeServerTag`, and `isBrokerTag` to reduce code redundancy.

Reply:
Simplified